recipes/configs/llama2/7B_full_low_memory.yaml

```diff
@@ -53,7 +53,10 @@ batch_size: 2
 epochs: 3
 optimizer:
   _component_: bitsandbytes.optim.PagedAdamW
-  lr: 2e-5
+  lr: 1e-5
+lr_scheduler:
+  _component_: torchtune.modules.get_cosine_schedule_with_warmup
+  num_warmup_steps: 1
```

Comment:
I saw your commit about needing to do this to pass the test. Which test is that referring to? I don't think we wanna make this change to our configs unless we know the results will be better, plus it's kinda weird to say we are using a scheduler "with warmup" when the warmup is just one step

Reply:
> I saw your commit about needing to do this to pass the test. Which test is that referring to? I don't think we wanna make this change to our configs unless we know the results will be better, plus it's kinda weird to say we are using a scheduler "with warmup" when the warmup is just one step

It should be `test_loss` of `tests/recipes/test_full_finetune_single_device.py`. So when I ran the test, the increase of the warm up steps would result higher loss tolerance of loss values than expected loss discrepancy. 
`torch.testing.assert_close(
            loss_values, expected_loss_values, rtol=1e-4, atol=1e-4
        )` 
I suspected that might due to warm up steps. However, if I change the tolerance the value might be too high. It might even reach 1e-1 level. However when warmup step is large we are effectively have a much smaller lr. Observed from my own training this might cause initial loss to have large differences. As we can see from my own training the initial loss have like 2e-1 difference but the difference would shrink with more training steps after warmup finished. I havent find a good way to resolve this though. Do you have any suggestions? I hope it isnt some bug from my code causing the tolerance difference thats' why I tested it with 1 warmup step...

<img width="719" alt="Screenshot 2024-09-05 at 7 00 44 AM" src="https://github.com/user-attachments/assets/04181c06-3264-494a-a15b-9bd434535ed1">

Reply:
@user074 ah I see, I think I was a bit confused about the exact issue but it makes sense now. Note that our recipe tests actually use test-specific overrides; that way we can decouple the configs we use for real fine-tunes from what we use for testing. You can see the overrides [here](https://github.com/pytorch/torchtune/blob/d31649e75e51cf15dbdca004288c32550e8f7f8d/tests/recipes/test_full_finetune_single_device.py#L36-L50) -- I would suggest just adding an override like `lr_scheduler=torch.optim.ConstantLr` there. Hopefully that'll fix things!

Reply:
Do you know how to deal with 'num_warmup_steps' when I set to ConstantLr. It seems to give error due to this extra argument. But I couldnt find a way to remove this keyword argument from test script. model_config does not contain it explicitly so I cannot remove.

Reply:
Still have problem with the larger loss with the ConstantLR. Kinda odd since warmup steps shouldnt have any affect now. What is left is OptimizerInBackwardWrapper behavior. Maybe I implemented something wrong there?

<img width="410" alt="Screenshot 2024-09-07 at 11 00 10 AM" src="https://github.com/user-attachments/assets/bd291ecd-d55b-4035-b734-8e207135ed2a">
<img width="586" alt="Screenshot 2024-09-07 at 11 09 33 AM" src="https://github.com/user-attachments/assets/97fea582-6bb3-49c9-828c-de6e3de5b972">

Reply:
I noticed you changed the learning rate in the config? Could that be it?

Reply:
> I noticed you changed the learning rate in the config? Could that be it?

Not really since the test recipe overrides the lr to 2e-5. I also just tested the 2e-5 as the original lr. It is the same problem with loss discrepancy.

Reply:
Looks to be working on the build machines. Could you push your latest changes, even if they break the tests, and I can check it out?

Reply:
Agree it'd be helpful to just push the changes and see what's happening. But also if using ConstantLR make sure you're setting `factor=1`